src/Microsoft.Azure.WebJobs.Host/Hosting/DefaultStartupTypeLocator.cs

```diff
@@ -9,18 +9,18 @@
 namespace Microsoft.Azure.WebJobs.Hosting
 {
     /// <summary>
-    /// An implementation of an <see cref="IWebJobsStartupTypeDiscoverer"/> that locates startup types
+    /// An implementation of an <see cref="IWebJobsStartupTypeLocator"/> that locates startup types
     /// configured in the entry point assembly using the <see cref="WebJobsStartupAttribute"/>.
     /// </summary>
-    internal class DefaultStartupTypeDiscoverer : IWebJobsStartupTypeDiscoverer
+    internal class DefaultStartupTypeLocator : IWebJobsStartupTypeLocator
```

Comment:
We already use "locator" terminology for existing interfaces like ITypeLocator @fabiocav

Reply:
This is a pretty impactful change and would, again, invalidate all extensions installed.

I was aware of the other `Locators`, but given the role of this interface, thought it made sense to use the name it had. Either way, I don't think we can make this change now as it would break A LOT of stuff.

Reply:
This locator interface is just used by the SDK and Functions internally - extensions just use the attribute. So after pulling the updated SDK into functions and absorbing this change, all extensions work as is, right? Also, for GA we'll be revving all the extensions once more anyways, targeting the final SDK.

Reply:
Yes, this was my bad. Somehow I got IWebJobsStartup in my mind. These will be fine. I realized on my way home and was going to delete the comment, but you beat me to it.